exporters/trace/ocagent/src/test/java/io/opencensus/exporter/trace/ocagent/OcAgentTraceServiceRpcHandlersTest.java

```diff
@@ -0,0 +1,186 @@
+import io.opencensus.proto.agent.trace.v1.ExportTraceServiceRequest;
+import io.opencensus.proto.agent.trace.v1.TraceServiceGrpc;
+import io.opencensus.proto.agent.trace.v1.TraceServiceGrpc.TraceServiceStub;
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.config.TraceParams;
+import java.io.IOException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+/**
+ * Unit tests for {@link OcAgentTraceServiceConfigRpcHandler} and {@link
+ * OcAgentTraceServiceExportRpcHandler}.
+ */
+@RunWith(JUnit4.class)
+public class OcAgentTraceServiceRpcHandlersTest {
+
+  @Mock private TraceConfig mockTraceConfig;
+
+  private Server server;
+  private FakeOcAgentTraceServiceGrpcImpl traceServiceGrpc;
+  private String serverName;
+
+  private static final Node NODE =
+      Node.newBuilder()
+          .setLibraryInfo(LibraryInfo.newBuilder().setLanguage(Language.JAVA).build())
+          .build();
+  private static final io.opencensus.proto.trace.v1.TraceConfig TRACE_CONFIG_DEFAULT_PROTO =
+      TraceProtoUtils.toTraceConfigProto(TraceParams.DEFAULT);
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+    Mockito.doReturn(TraceParams.DEFAULT).when(mockTraceConfig).getActiveTraceParams();
+    Mockito.doNothing().when(mockTraceConfig).updateActiveTraceParams(any(TraceParams.class));
+
+    traceServiceGrpc = new FakeOcAgentTraceServiceGrpcImpl();
+    serverName = InProcessServerBuilder.generateName();
+    server =
+        InProcessServerBuilder.forName(serverName)
+            .directExecutor() // directExecutor is fine for unit tests
+            .addService(traceServiceGrpc)
+            .build()
+            .start();
+  }
+
+  @After
+  public void tearDown() {
+    if (server != null && !server.isTerminated()) {
+      server.shutdown();
+    }
+  }
+
+  @Test
+  public void config_CreateAndSend() throws InterruptedException {
+    // Config RPC handler needs to be running in another thread.
+    Runnable configRunnable =
+        new Runnable() {
+          @Override
+          public void run() {
+            TraceServiceStub stub = getStub(serverName);
+            OcAgentTraceServiceConfigRpcHandler configRpcHandler =
+                OcAgentTraceServiceConfigRpcHandler.create(stub, mockTraceConfig);
+            assertThat(configRpcHandler.isCompleted()).isFalse(); // connection should succeed
+            configRpcHandler.sendInitialMessage(NODE); // this will block this thread
+          }
+        };
+
+    Thread configThread = new Thread(configRunnable);
+    configThread.setDaemon(true);
+    configThread.setName("TestConfigRpcHandlerThread");
+    configThread.start();
+
+    while (traceServiceGrpc.getCurrentLibraryConfigs().isEmpty()) {
```

Comment:
Make it deterministic and use a conditional variable.

Reply:
Done.